packages/contracts/contracts/ChugSplashManager.sol

```diff
@@ -373,6 +423,73 @@ contract ChugSplashManager is Owned {
         }
     }
 
+    /**
+     * @notice Allows an executor to post a bond of `executorBondAmount` to claim the sole right to
+     *         execute actions for a bundle over a period of `executionLockTime`. Only the first
+     *         executor to post a bond gains this right. Executors must finish executing the bundle
+     *         within `executionLockTime` or else their bond is forfeited to this contract and
+     *         another executor may claim the bundle. Note that this strategy creates a PGA for the
+     *         transaction to claim the bundle but removes PGAs during the execution process.
+     *
+     * @param _bundleId ID of the bundle being claimed.
+     */
+    function claim(bytes32 _bundleId) external payable {
+        require(
+            executorBondAmount == msg.value,
+            "ChugSplashManager: incorrect executor bond amount"
+        );
+        ChugSplashBundleStatus status = bundles[_bundleId].status;
+        require(
+            status == ChugSplashBundleStatus.APPROVED,
+            "ChugSplashManager: bundle is not active"
+        );
+
+        ClaimedBundle storage bundle = claimedBundles[_bundleId];
+
+        uint256 executionDeadline = bundle.timeClaimed + executionLockTime;
+        require(
+            block.timestamp > executionDeadline,
+            "ChugSplashManager: bundle already claimed by an executor"
+        );
+
+        address prevSelectedExecutor = bundle.selectedExecutor;
+
+        bundle.timeClaimed = block.timestamp;
+        bundle.selectedExecutor = msg.sender;
+
+        emit BundleClaimed(_bundleId, msg.sender, prevSelectedExecutor);
```

Comment:
I think we need to propagate this by sending to `announce` on the `ChugSplashRegistry`